graphql/schemabuilder/reflect.go

```diff
@@ -507,6 +518,9 @@ func (sb *schemaBuilder) buildFunction(typ reflect.Type, fun reflect.Value) (*gr
 			var result interface{}
 			if hasRet {
 				result = out[0].Interface()
+				if out[0].Kind() == reflect.Ptr && out[0].IsNil() && !m.MarkedNullable {
```

Comment:
^ I was thinking about the same thing and I wonder if we can just return pointer type for nullable type and a non-pointer type for non-nullable. (this is what we do today)
Are there other benefits of this PR's approach that I may have overlooked that can't be achieved by what we're doing today?